src/main/java/org/cactoos/text/CapitalizedText.java

```diff
@@ -53,15 +53,15 @@ public CapitalizedText(final Text text) {
     @Override
     public String asString() throws IOException {
         return new JoinedText(
-            new TransformedIterable<String, Text>(
+            new TransformedIterable<Text, Text>(
                 new SplitText(this.origin),
-                new Func<String, Text>() {
+                new Func<Text, Text>() {
                     @Override
-                    public Text apply(final String input) throws Exception {
+                    public Text apply(final Text input) throws Exception {
```

Comment:
I didn't have the secondary constructor in mind, but if you do it that way,  the primary one would need to be private.
Anyway, why the second ctor? you would never want to give it a different capitalize function?! Just keep the first ctor and remove the second one.

And if we could replace this hideous java extra verbosity with lambdas
```
public class CapitalizedText implements Text {
    private final Text origin;
    private final Func<Text, Text> func;

    public CapitalizedText(final Text text) {
        this.origin = text;
        this.func = input -> new FormattedText(
            "%s%s",
            input.asString().substring(0, 1).toUpperCase(),
            input.asString().substring(1)
        );
   
    }
```
i would be perfectly satisfied without anything left to complain